quasar-utils/src/main/groovy/net/corda/plugins/quasar/QuasarPlugin.groovy

```diff
@@ -85,6 +95,26 @@ class QuasarPlugin implements Plugin<Project> {
             }
             dependencies.add(quasarDependency)
         }
+
+        def javacPluginConfiguration = project.configurations.create(JAVAC_PLUGIN)
+        javacPluginConfiguration.withDependencies { dependencies ->
+            if(extension.enableJavacPlugin.get()) {
+                def cls = getClass()
+                def resourceName = cls.name.replace('.', '/') + ".class"
+                def classUrl = cls.classLoader.getResource(resourceName)
+                if (classUrl.protocol.startsWith("jar")) {
+                    def path = classUrl.toString()
+                    String manifestPath = path.substring(0, path.lastIndexOf("!") + 1) +
+                            "/META-INF/MANIFEST.MF"
+                    Manifest manifest = new Manifest(new URL(manifestPath).openStream())
+                    Attributes attr = manifest.getMainAttributes()
+                    String version = attr.getValue("version")
+                    dependencies.add(project.dependencies.create("net.corda.plugins:javac-plugin:$version") {
+                        it.transitive = false
+                    })
+                }
+            }
+        }
```

Comment:
If we expect to download the compiler plugin as a regular project dependency then why does it also need to be an `implementation` dependency of `quasar-utils`?

Reply:
actually my idea was to simply bundle it with `quasar-utils` for simplicity

Reply:
I actually had forgotten of this bit of code which looks really horrible.. I think I will simply include the compiler plugin in quasar-utils jar